### ISSUE TITLE
[Bug]: fetch failed for source

### ISSUE THREAD

**Issue by mmstano** [mampfes/hacs_waste_collection_schedule#2420]
### I Have A Problem With:

A specific source

### What's Your Problem

I’ve been using this integration since the yaml-only days and have never experienced an issue until yesterday when the integration stopped working for me. No information is being pulled from Republic to update my sensors. I’ve tried uninstalling/reinstalling to no avail. I’ve tried updating my value templates for the sensors. I’ve also tried migrating from yaml config to the GUI config and receive the same error. This problem leads me to think the issue happened because of something Republic may have changed on their end. 

I’ve tried updating to the master branch but the version selector keeps reverting to the latest stable (I’m assuming this is an issue of HACS unrelated to WCS, as I’m experiencing this with all of my HACS custom components)

### Source (if relevant)

Republic Services

### Logs

```Shell
Logger: waste_collection_schedule.source_shell
Source: custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py:147
integration: Waste Collection Schedule (documentation)
First occurred: 8:38:44 AM (1 occurrences)
Last logged: 8:38:44 AM

fetch failed for source Republic Services: Traceback (most recent call last): File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 145, in fetch entries = self._source.fetch() ^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/republicservices_com.py", line 106, in fetch dt = datetime.strptime( ^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/_strptime.py", line 554, in _strptime_datetime tt, fraction, gmtoff_fraction = _strptime(data_string, format) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/_strptime.py", line 333, in _strptime raise ValueError("time data %r does not match format %r" % ValueError: time data '2024-01-01T00:00:00.000Z' does not match format '%Y-%m-%dT00:00:00.0000000Z'
```

### Relevant Configuration

```YAML
waste_collection_schedule:
  sources:
    - name: republicservices_com
      args: 
        street_address: “ADDRESS”
        method: 1

sensor:
 - platform: waste_collection_schedule
    name: recycle_date
    types:
      - Recycle
    value_template: '{{value.date.strftime("%m/%d/%Y")}}'
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [X] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [ ] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by mmstano** [comment]
I managed to force an update to master using the update service call, but still get the fetch error.

**Comment by 5ila5** [comment]
Fixed in #2422

This will be part of the next release or is available now as master (git) version (3-dot menu -> redownload -> change version to master -> fully restart HA (this will increase update cycle by a lot)
